health/warnings.go

```diff
@@ -0,0 +1,216 @@
+	Severity:  SeverityHigh,
+	DependsOn: []*Warnable{NetworkStatusWarnable},
+	Text: func(args map[string]string) string {
+		return "Tailscale couldn't listen for incoming UDP connections."
+	},
+}, WithConnectivityImpact())
+
+// mapResponseTimeoutWarnable is a Warnable that warns the user that Tailscale hasn't received a network map from the coordination server in a while.
+var mapResponseTimeoutWarnable = Register(&Warnable{
+	Code:      "mapresponse-timeout",
+	Title:     "Network map response timeout",
+	Severity:  SeverityMedium,
+	DependsOn: []*Warnable{NetworkStatusWarnable},
+	Text: func(args map[string]string) string {
+		return fmt.Sprintf("Tailscale hasn't received a network map from the coordination server in %s.", args["Duration"])
+	},
+})
+
+// tlsConnectionFailedWarnable is a Warnable that warns the user that Tailscale could not establish an encrypted connection with a server.
+var tlsConnectionFailedWarnable = Register(&Warnable{
+	Code:      "tls-connection-failed",
+	Title:     "Encrypted connection failed",
+	Severity:  SeverityMedium,
+	DependsOn: []*Warnable{NetworkStatusWarnable},
+	Text: func(args map[string]string) string {
+		return fmt.Sprintf("Tailscale could not establish an encrypted connection with '%s': %v", args["ServerName"], args["Error"])
+	},
+})
+
+// magicsockReceiveFuncWarnable is a Warnable that warns the user that one of the Magicsock functions is not running.
+var magicsockReceiveFuncWarnable = Register(&Warnable{
+	Code:     "magicsock-receive-func-error",
+	Title:    "MagicSock function not running",
+	Severity: SeverityMedium,
+	Text: func(args map[string]string) string {
+		return fmt.Sprintf("The MagicSock function %s is not running. You might experience connectivity issues.", args["FunctionName"])
+	},
+})
+
+// testWarnable is a Warnable that is used within this package for testing purposes only.
+var testWarnable = Register(&Warnable{
+	Code:     "test-warnable",
+	Title:    "Test warnable",
+	Severity: SeverityLow,
+	Text: func(args map[string]string) string {
+		return args["Error"]
+	},
+})
+
+// applyDiskConfigWarnable is a Warnable that warns the user that there was an error applying the envknob config stored on disk.
+var applyDiskConfigWarnable = Register(&Warnable{
+	Code:     "apply-disk-config",
+	Title:    "Could not apply configuration",
+	Severity: SeverityMedium,
+	Text: func(args map[string]string) string {
+		return fmt.Sprintf("An error occurred applying the Tailscale envknob configuration stored on disk: %v", args["Error"])
+	},
+})
+
+// controlHealthWarnable is a Warnable that warns the user that the coordination server is reporting an health issue.
+var controlHealthWarnable = Register(&Warnable{
+	Code:     "control-health",
+	Title:    "Coordination server reports an issue",
+	Severity: SeverityMedium,
+	Text: func(args map[string]string) string {
+		return fmt.Sprintf("The coordination server is reporting an health issue: %v", args["Error"])
```

Comment:
I think the args should all be lowercased. Maybe camelCased. Or hyphenated. But lowercase regardless.

Reply:
Done, and moved all of these to constants in a dedicated file